lxd/migration/migration_volumes.go

```diff
@@ -0,0 +1,191 @@
+package migration
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/lxc/lxd/lxd/operations"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/ioprogress"
+	"github.com/lxc/lxd/shared/units"
+)
+
+// Type represents the migration transport type. It indicates the method by which the migration can
+// take place and what optional features are available.
+type Type struct {
+	FSType   MigrationFSType
+	Features []string
+}
+
+// VolumeSourceArgs represents the arguments needed to setup a volume migration source.
+type VolumeSourceArgs struct {
+	Name          string
+	Snapshots     []string
+	MigrationType Type
+	LXDExecPath   string
```

Comment:
You shouldn't need that. We should have access to `State` in all of the storage structs and you can access the LXD exec path through `State.OS.ExecPath`.

Reply:
As discussed, I will remove this and expose the `state.State` field to the underlying storage driver so rsync can access state.OS.ExecPath directly.